File: program/check-anagram-string/CheckAnagramString.java
import java.util.Scanner;
import java.util.Arrays;

public class CheckAnagramString {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Get the strings from the user
        System.out.print("Enter the first word: ");
        String word1 = scan.next();

        System.out.print("Enter the second word: ");
        String word2 = scan.next();

        // Check if the strings are anagrams and print a message to user
        if(isAnagram(word1, word2)) {
            System.out.println("The words are anagrams.");
        }
        else {
            System.out.println("The words are not anagrams.");
        }

        scan.close();
    }

    /**
     * Checks if two strings are anagrams
     * @param word1 String
     * @param word2 String
     * @return Boolean
     */
    public static boolean isAnagram(String word1, String word2) {
        // Convert both strings to lowercase
        word1 = word1.toLowerCase();
        word2 = word2.toLowerCase();

        // Remove all non-alphabetic characters
        word1 = word1.replaceAll("[^a-z]", "");
        word2 = word2.replaceAll("[^a-z]", "");

        // If the strings have not the same length, there are not anagrams
        if(word1.length() != word2.length()) {
            return false;
        }

        // Sort both strings
        char[] charArray1 = word1.toCharArray();
        char[] charArray2 = word2.toCharArray();
        Arrays.sort(charArray1);
        Arrays.sort(charArray2);

        // Compare the sorted strings. If there are equal, the strings are anagrams.
        return Arrays.equals(charArray1, charArray2);
    }
}

File: program/check-anagram-string/check-anagram-string.jl
function is_anagram(str1::String, str2::String)
    # Remove any spaces and convert both strings to lowercase
    str1 = replace(lowercase(str1), r"\s+" => "")
    str2 = replace(lowercase(str2), r"\s+" => "")

    # Check if the length of both strings is the same
    if length(str1) != length(str2)
        return false
    end

    # Sort the characters in both strings and check if they are equal
    if sort(collect(str1)) == sort(collect(str2))
        return true
    else
        return false
    end
end

# Example usage
println("Enter 2 Strings") #take input
str1 = readline()
str2 = readline()

if is_anagram(str1, str2)
    println("Anagram Strings")
else
    println("Not Anagram Strings")
end